### PR TITLE
Update non-functional requirements and glossary

### LINKED ISSUE #4: Update developer guide

### PR DESCRIPTION
**Pull request by pualixue**
Resolves #4

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -312,16 +312,28 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 
 ### Glossary
 
+* **Appointment**: Arrangement to meet between a doctor and a patient
+* **CLI**: Command Line Interface
+* **Contact**: Records of personal information of a patient/doctor
+* **GUI**: Graphical User Interface
 * **Mainstream OS**: Windows, Linux, Unix, OS-X
+* **Patient**: The individual that visits the clinic
 * **Private contact detail**: A contact detail that is not meant to be shared with others
```
Comment (jiayi1129) [inline docs/DeveloperGuide.md:335]: Do we use the term "private contact detail" in the other parts of the developer guide? If it's not applicable to our current project perhaps we can remove this? Let me know what you think!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -312,16 +312,28 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Non-Functional Requirements
 
-1.  Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
-2.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+1. Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
+2. Should work independent of network connection.
+3. Should respond to users' commands within two seconds.
+4. Should be able to store 5000 unique entries of patients and doctors.
```
Comment (jiayi1129) [inline docs/DeveloperGuide.md:318]: How did you come up with the number 5000? I think its a good estimate but perhaps you can share your considerations.

**Review by jiayi1129 — COMMENTED** [review]
Overall looks good to me! Some minor nitpicks

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (pualixue) [reply docs/DeveloperGuide.md:335]: Yes it is mentioned in the user stories #15.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (pualixue) [reply docs/DeveloperGuide.md:318]: I just did a very rough estimate by dividing the total population by the number of clinics, and then increased the estimate to cater for more patients.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -312,16 +312,28 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 ### Glossary
 
+* **Appointment**: Arrangement to meet between a doctor and a patient
+* **CLI**: Command Line Interface
+* **Contact**: Records of personal information of a patient/doctor
+* **GUI**: Graphical User Interface
 * **Mainstream OS**: Windows, Linux, Unix, OS-X
+* **Patient**: The individual that visits the clinic
 * **Private contact detail**: A contact detail that is not meant to be shared with others
+* **Risk profile**: The health status and severity of the condition of a patient
```
Comment (Haozeee) [inline docs/DeveloperGuide.md:336]: I think this may need to be updated depending on how we implement risk profiles. We can discuss it on Sunday.

**Review by Haozeee — APPROVED** [review]
Overall looks good.

**Pull request merged**